src/mount_efs/__init__.py

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env -S python -Es
```

Comment:
From what i can tell, the version of `/usr/bin/env` that ships with the distros we support for `amazon-efs-utils` does not actually support the `-S` option. This causes the mount helper and watchdog to not launch properly. Simply removing the `-S` seems to cause other issues, but that may be an artifact of `python -Es` not being parsed correctly without the `-S` option. 

Is there an alternative way you could add `python -Es` without `/usr/bin/env -S`? Or a way to properly update `env`? We currently support `amazon-efs-utils` on the following distros:

* Amazon Linux
* Amazon Linux 2
* CentOS 7
* RHEL 7
* Debian 9
* Ubuntu 16.04

Reply:
Oh indeed the `-S` argument of /usr/bin/env is a rather recent addition to coreutils version 8.30 (released 2018-07-01).

This shebang business is a rather dark corner on Linux. The kernel does only minimal command line processing for the she-bang. It separates the interpreter from the rest of the command line and passes the command line as a single parameter to the interpreter, ignoring any whitespace. The `-S` parameter to `/usr/bin/env` triggers userspace logic to split up the single argument received into separate ones.

So I see various approaches to deal with this:

- Drop `/usr/bin/env` and simply use `#!/usr/bin/python -Es` as shebang. This works if all of the targeted distributions install python in `/usr/bin/python` which I guess is pretty common.
- Determine the correct location of `python` during build time and fill in the result as shebang in the Python scripts.
- Use a wrapper shell script with shebang `#!/bin/sh` that in turn runs `exec /usr/bin/env python -Es /path/to/python-script`. This would require changes in the installation logic of efs-utils.
- There's also the possiblity to use a crude hack:
```
#!/bin/sh

""":";exec /usr/bin/env python -Es $0;":"""

import os
import sys
```

The hack abuses the python docstring, to be interpreted by the regular shell, restarting itself as the python interpreter with the desired lookup semantics and command line arguments.